jerry-core/parser/js/js-parser-expr.c

```diff
@@ -587,7 +587,10 @@ parser_parse_class (parser_context_t *context_p, /**< context */
     }
   }
 
-  if (context_p->token.type == LEXER_KEYW_EXTENDS)
+  bool create_class_env = (bool) (context_p->token.type == LEXER_KEYW_EXTENDS
+                                  || context_p->status_flags & PARSER_CLASS_HAS_SUPER);
```

Comment:
Hmm. I think this should rather be written as
```c
(context_p->token.type == LEXER_KEYW_EXTENDS) 
|| (bool) (context_p->status_flags & PARSER_CLASS_HAS_SUPER)
```
The == results a bool, the || results a bool. It's only the & that needs casting.